xarray/backends/zarr.py

```diff
@@ -63,13 +63,13 @@ def encode_zarr_attr_value(value):
 class ZarrArrayWrapper(BackendArray):
     __slots__ = ("datastore", "dtype", "shape", "variable_name", "_array")
 
-    def __init__(self, variable_name, datastore):
+    def __init__(self, variable_name, zarr_array, datastore):
```

Comment:
```suggestion
    def __init__(self, zarr_array):
```
Fairly sure this should work.

Reply:
Test this properly. I've not had good results doing this trick on a different datastore.

Reply:
I'm not completely sure how the zarr backend is implemented, but in my case after `xr.open_dataset()` the datastore is closed and not needed anymore( until .compute()). 

But in this case you're sending a reference from that store to the backendarray which will be lost once the store is closed.

I've gone the opposite way, in `get_variables` only the required keys are sent. Then only once the backendarray `__getitem__ `is triggered the arrays are loaded.

I noticed this quite quickly once I played around with the dataset and loaded some variables and if you haven't noticed it yet maybe the zarr backend does some interesting tricks to avoid the closed references.

Reply:
Yerah not sure. We do have distributed tests for reading zarr so presumably it's all OK